gpbft/api.go

```diff
@@ -2,11 +2,16 @@ package gpbft
 
 import "time"
 
-// Receives EC chain values.
+// ChainReceiver defines the interface for receiving updates to an Expected Consensus (EC) chain.
+// Implementers of this interface can act upon received EC chain data, potentially considering the
+// given chain as the expected consensus chain in the current GPBFT instance.
 type ChainReceiver interface {
-	// Receives a new EC chain, and notifies the current instance if it extends its current acceptable chain.
-	// This modifies the set of valid values for the current instance.
-	ReceiveECChain(chain ECChain) error
+	// ReceiveECChain processes an incoming EC chain update. It assesses whether the new chain
+	// extends or is compatible with the currently accepted chain maintained by the instance.
+	//
+	//Returns a boolean indicating whether the chain was accepted. An error is returned if the
+	// given chain is invalid.
+	ReceiveECChain(chain ECChain) (bool, error)
```

Comment:
We could use sentinel error instead. Is there a reason not to do that?

Reply:
Is there a reason not to use a boolean flag considering the pattern is already present in the gpbft API? 

Either works. My only desirable is consistency.

Reply:
In the interest of quicker turnaround I have updated the code to use sentinel error instead.

Please let me know if there is anything else that I might have missed.